packages/vite/src/assets.ts

```diff
@@ -66,7 +68,7 @@ export function assets(): Plugin {
             this.emitFile({
               type: 'asset',
               source: readFileSync(join(pkg.root, path)),
-              fileName: dest.slice(1),
+              fileName: join('.', dest),
```

Comment:
interesting, the result for `join('.', './a-file')` or `join('.', '/a-file')` is always `'a-file'`, nice

Reply:
I changed the code to `posix.resolve(‘/’, dest).slice(1)` as suggested by Ed, it works as expected with my test setup (no problem in the middleware either) 👍